projects/data-grid/src/lib/store/data-grid.ts

```diff
@@ -379,7 +426,7 @@ const rowIndexesAndPaginationReducer = createReducer(initialGridState, on(
   GridActions.changePageNumber,
   GridActions.deleteRow,
   GridActions.addRow,
-  recalculateRowIndexesAndPagination
+  R.compose(recalculateSelectedRowsIndexes, recalculateRowIndexesAndPagination)
```

Comment:
It's better to have an `updateGridRowsHandler` that will recalculate row indexes, selected rows and pagination.